v2/cmd/wails/flags/common.go

```diff
@@ -1,5 +1,103 @@
 package flags
 
+import (
+	"fmt"
+	"os"
+	"runtime"
+	"strings"
+
+	"github.com/leaanthony/slicer"
+	"github.com/pterm/pterm"
+	"github.com/wailsapp/wails/v2/internal/system"
+)
+
 type Common struct {
 	NoColour bool `description:"Disable colour in output"`
 }
+
+type Target struct {
+	Platform string
+	Arch     string
+}
+
+func defaultTarget() Target {
+	defaultPlatform := os.Getenv("GOOS")
+	if defaultPlatform == "" {
+		defaultPlatform = runtime.GOOS
+	}
+	defaultArch := os.Getenv("GOARCH")
+	if defaultArch == "" {
+		if system.IsAppleSilicon {
+			defaultArch = "arm64"
+		} else {
+			defaultArch = runtime.GOARCH
+		}
+	}
+
+	return Target{
+		Platform: defaultPlatform,
+		Arch:     defaultArch,
+	}
+}
+
+type TargetsCollection []Target
+
+func (c TargetsCollection) MacTargetsCount() int {
+	count := 0
+
+	for _, t := range c {
+		if strings.HasPrefix(t.Platform, "darwin") {
+			count++
+		}
+	}
+
+	return count
+}
+
+func (t Target) String() string {
+	if t.Arch != "" {
+		return fmt.Sprintf("%s/%s", t.Platform, t.Arch)
+	}
+
+	return t.Platform
+}
+
+func parseTargets(platform string) TargetsCollection {
+	allowedPlatforms := map[string]bool{
+		"windows": true,
+		"linux":   true,
+		"darwin":  true,
+	}
+
+	if !allowedPlatforms[platform] {
```

Comment:
This checks breaks the ability to cross compile to another platform with a specific architecture (e.g. `--platform windows/arm64` or `--platform darwin/universal`), because the argument `platform` is the arg cli `--platform` arg.
It also prevents the ability to compile for multiple platforms `--platform windows/am64,windows/arm64`.

Reply:
PR to resolve this one:
https://github.com/wailsapp/wails/pull/3245